src/mixins/treeitem.js

```diff
@@ -0,0 +1,82 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import orderBy from 'lodash/orderBy'
+
+const treeitem = {
+  methods: {
+    /**
+     * Sort children based on the type of the node.
+     *
+     * Each type may have a different sorting strategy. For example,
+     * for nodes of type workflow, we know our children will be cycle points.
+     * So we may sort them by their `id`, and in ascending or descending
+     * order.
+     *
+     * Ditto for other types. This way we can control how to sort the children
+     * nodes of each node in the UI.
+     *
+     * Some elements are sorted in the GraphQL query, on the server-side. While
+     * some times these elements may have to be re-sorted, the best case
+     * scenario is that the elements are already in order, and the JS sorting
+     * algorithm doesn't have to sort anything.
+     *
+     * NOTE: the Array.sort algorithm implementation is browser-engine dependent,
+     * and also type dependent. One browser may use quick-sort for numeric arrays,
+     * and merge-sort for contiguous arrays of non-numeric types, and selection-sort for
+     * all the other cases (source: https://stackoverflow.com/questions/234683/javascript-array-sort-implementation
+     * and http://trac.webkit.org/browser/trunk/Source/JavaScriptCore/runtime/ArrayPrototype.cpp?rev=138530#L647).
+     *
+     * @param type {string} - node type (e.g. 'workflow', 'cyclepoint', etc)
+     * @param children {Array} - node children (e.g. list of jobs of a task-proxy)
+     * @returns {Array} - sorted children
+     */
+    sortedChildren (type, children) {
+      if (children && children.length !== 0) {
+        switch (type) {
+        case 'workflow': {
+          // sort workflow children (cycle-points)
+          // sort by id descending, so '20100102' comes before '20100101'
+          return orderBy(children, ['id'], 'desc')
+        }
+        case 'cyclepoint': {
+          // sort cycle point children (family-proxies, and task-proxies)
+          // first we sort by type ascending, so 'family-proxy' types come before 'task-proxy'
+          // then we sort by node name ascending, so 'bar' comes before 'foo'
+          return orderBy(children, [child => child.type, child => child.node.name.toLowerCase()], ['asc', 'asc'])
+        }
+        case 'family-proxy': {
+          // sort family-proxy children (family-proxies, and task-proxies)
+          // first we sort by type ascending, so 'family-proxy' types come before 'task-proxy'
+          // then we sort by node name ascending, so 'bar' comes before 'foo'
+          return orderBy(children, [child => child.type, child => child.node.name.toLowerCase()], ['asc', 'asc'])
```

Comment:
Follow on:

It might be nice to have the families appear before the tasks similarly to how some file listings put directories before files.

Because of the Cylc naming convention of lower-case for tasks, upper-case for families removing the `toLowerCase` would achieve this.

Do we think people would universally prefer this, should it be configurable?

Reply:
>It might be nice to have the families appear before the tasks similarly to how some file listings put directories before files.

That's how it should be working now. The family proxies should be displayed before the task proxies (because we order by 2 fields, first type, and then name).

>Do we think people would universally prefer this, should it be configurable?

Good question. Will leave that for follow-up question with @cylc/core

Reply:
I don't see a great need to make that configurable.  A possible future change if people want it.

Reply:
I think people should be happy enough with this.